### PR TITLE
fix some api error displaying

### LINKED ISSUE #530: [BUG] Using latin characters in ASN search displays object reference
**Describe the bug**

When using latin characters in the ASN search a `[object Object]` reference shows in the error message

**To Reproduce**

1. Start paperless
2. Go to documents
3. Select ASN in the drop-down
4. Search with non number characters

**Expected behavior**

Non-technical error message displayed. Maybe even telling the user that only latin numbers allowed for the ASN search

**Screenshots**

image

**Webserver logs**

```bash
webserver_1  | [2022-03-26 10:21:34,232] [WARNING] [django.request] Bad Request: /api/documents/
```

**Relevant information**
 - Host OS of the machine running paperless: Ubuntu 21.04
 - Browser [e.g. chrome, safari]: firefox
 - Version [e.g. 1.0.0]: 1.6.0
 - Installation method: docker

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

This PR addresses a couple of areas where errors were not being displayed properly. It also adds localization to one of the error messages.

Screen Shot 2022-04-02 at 5 56 08 PM

Note the above error is different than if the view title is too long:
Screen Shot 2022-04-02 at 5 55 45 PM

Screen Shot 2022-04-02 at 5 11 18 PM

Part of #601
Closes #530

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#code-formatting-with-pre-commit-hooks).
- [ ] ~~I have made corresponding changes to the documentation as needed.~~
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Review by crazychatting — APPROVED** [review]
lgtm

**Comment by shamoon** [comment]
Im going to incorporate https://github.com/paperless-ngx/paperless-ngx/issues/530#issuecomment-1086729440 into this when I have a second, will keep as draft until then.

**Comment by shamoon** [comment]
Ok I sorted this out, was actually slightly confusing. I updated the above but TLDR there are two separate errors, the title can be too long but and / or the filter rule can be.

I also added the other issue which was related but not the same.

File: src-ui/src/app/components/document-list/document-list.component.ts [hunk 1]
```diff
@@ -243,8 +243,12 @@ export class DocumentListComponent implements OnInit, OnDestroy, AfterViewInit {
               $localize`View "${savedView.name}" created successfully.`
             )
           },
-          error: (error) => {
-            modal.componentInstance.error = error.error
+          error: (apiError) => {
```
Comment (shamoon) [inline src-ui/src/app/components/document-list/document-list.component.ts:246]: TODO

**Review by qcasey — APPROVED** [review]
Works well.

**Pull request merged**